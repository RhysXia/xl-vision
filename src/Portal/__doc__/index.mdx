# Portal

可以灵活方便地将子组件挂载到其他位置的组件

:::demo 基础用法

基础用法

```jsx
import React from 'react'
import { Portal, Button } from 'xl-vision'

export default () => {
  const [node, setNode] = React.useState()

  React.useEffect(() => {
    const portal = (
      <Portal getContainer='.container2'>
        <Button theme='primary' className='content'>
          点击
        </Button>
      </Portal>
    )
    setNode(portal)
  }, [])

  const onClick = React.useCallback(() => {
    alert('You clicked container1!')
  }, [])

  return (
    <div>
      <div className='container1' onClick={onClick}>
        <p>container1</p>
        {node}
      </div>
      <div className='container2'>
        <p>container2</p>
      </div>
    </div>
  )
}
```

```scss
.container1,
.container2 {
  height: 100px;
  color: #fff;
  text-align: center;
  padding: 15px;
  border-radius: 4px;

  > p {
    font-size: 20px;
    font-weight: bold;
  }
}

.container1 {
  background-color: #8bc34a;
}

.container2 {
  margin-top: 16px;
  background-color: #e91e63;
}
```

:::

## Portal API

| 参数 | 说明 | 类型 | 可选值 | 默认值 | 是否必填 |
| --- | --- | --- | --- | --- | --- |
| getContainer | 指定挂载的父节点，支持字符串，节点实例或返回这些的函数 | string &#x007c; HTMLElement &#x007c; (() => string &#x007c; HTMLElement) | - | - | false |
| children | 子元素 | ReactNode | - | — | true |
