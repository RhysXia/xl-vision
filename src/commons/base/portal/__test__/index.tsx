import { mount } from 'enzyme'
import React from 'react'
import Portal from '..'

describe('Portal', () => {
  it('测试挂载到body中', () => {
    const wrapper = mount(
      <div>
        <Portal getContainer={() => document.body}>
          <div>123</div>
        </Portal>
      </div>
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('直接挂载在当前位置', () => {
    const wrapper = mount(
      <div>
        <Portal getContainer={() => null}>
          <div>123</div>
        </Portal>
      </div>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
