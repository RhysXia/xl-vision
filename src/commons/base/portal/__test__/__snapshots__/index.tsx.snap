// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portal 测试挂载到body中 1`] = `
<div>
  <Portal
    getContainer={[Function]}
  >
    <Portal
      containerInfo={
        <body>
          <div>
            123
          </div>
        </body>
      }
    >
      <div>
        123
      </div>
    </Portal>
  </Portal>
</div>
`;

exports[`Portal 直接挂载在当前位置 1`] = `
<div>
  <Portal
    getContainer={[Function]}
  >
    <div>
      123
    </div>
  </Portal>
</div>
`;
